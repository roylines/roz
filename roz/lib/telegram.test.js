const {send} = require('./telegram');
const {ssm} = require('./aws');
const random = require('random-int');
const {TelegramClient} = require('messaging-api-telegram');

jest.mock('await-sleep');
jest.mock('lambda-log');
jest.mock('messaging-api-telegram');
jest.mock('random-int');
jest.mock('./aws');

describe('message', () => {
  const mockTelegram = () => {
    ssm.getTelegramSecrets.mockResolvedValue({token: 'TOKEN', user: '42'});
    const connect = {sendMessage: jest.fn(), sendChatAction: jest.fn()};
    connect.sendMessage.mockResolvedValue();
    connect.sendChatAction.mockResolvedValue();
    TelegramClient.connect.mockReturnValue(connect);
    TelegramClient.connect.mockClear();
    return {connect: TelegramClient.connect, ...connect};
  };

  test('it should send expected rand 0', async () => {
    const {connect, sendMessage, sendChatAction} = mockTelegram();
    random.mockReturnValue(0);
    await send([['MESSAGE1-1', 'MESSAGE1-2'], ['MESSAGE2-1', 'MESSAGE2-2']]);
    expect(connect.mock.calls).toMatchSnapshot();
    expect(sendMessage.mock.calls).toMatchSnapshot();
    expect(sendChatAction.mock.calls).toMatchSnapshot();
    expect(random.mock.calls).toMatchSnapshot();
  });
  test('it should send expected rand 1', async () => {
    const {connect, sendMessage, sendChatAction} = mockTelegram();
    random.mockReturnValue(1);
    await send([['MESSAGE1-1', 'MESSAGE1-2'], ['MESSAGE2-1', 'MESSAGE2-2']]);
    expect(connect.mock.calls).toMatchSnapshot();
    expect(sendMessage.mock.calls).toMatchSnapshot();
    expect(sendChatAction.mock.calls).toMatchSnapshot();
    expect(random.mock.calls).toMatchSnapshot();
  });
});
