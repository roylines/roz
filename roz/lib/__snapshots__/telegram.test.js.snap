// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message it should send expected rand 0 1`] = `
Array [
  Array [
    "TOKEN",
  ],
]
`;

exports[`message it should send expected rand 0 2`] = `
Array [
  Array [
    "42",
    "MESSAGE1-1",
  ],
  Array [
    "42",
    "MESSAGE1-2",
  ],
]
`;

exports[`message it should send expected rand 0 3`] = `
Array [
  Array [
    "42",
    "typing",
  ],
]
`;

exports[`message it should send expected rand 0 4`] = `
Array [
  Array [
    0,
    1,
  ],
  Array [
    800,
    2500,
  ],
  Array [
    800,
    2500,
  ],
]
`;

exports[`message it should send expected rand 1 1`] = `
Array [
  Array [
    "TOKEN",
  ],
]
`;

exports[`message it should send expected rand 1 2`] = `
Array [
  Array [
    "42",
    "MESSAGE2-1",
  ],
  Array [
    "42",
    "MESSAGE2-2",
  ],
]
`;

exports[`message it should send expected rand 1 3`] = `
Array [
  Array [
    "42",
    "typing",
  ],
]
`;

exports[`message it should send expected rand 1 4`] = `
Array [
  Array [
    0,
    1,
  ],
  Array [
    800,
    2500,
  ],
  Array [
    800,
    2500,
  ],
  Array [
    0,
    1,
  ],
  Array [
    800,
    2500,
  ],
  Array [
    800,
    2500,
  ],
]
`;
